lib/cn/cn_tree.c

```diff
@@ -1700,16 +1647,7 @@ cn_tree_prepare_compaction(struct cn_compaction_work *w)
     w->cw_outv = outs;
     w->cw_vbmap = vbm;
     w->cw_drop_tombv = drop_tombs;
-    w->cw_hash_shift = 0;
-
-    if (n_outs > 1) {
-        uint bits = w->cw_tree->ct_fanout_bits;
-
-        if (cn_tree_get_khashmap(w->cw_tree))
-            bits = CN_KHASHMAP_SHIFT;
-
-        w->cw_hash_shift = bits * node->tn_loc.node_level;
-    }
+    w->cw_level = w->cw_node->tn_loc.node_level;
```

Comment:
Had to do this becaused some unit test sets up a cw struct with a nil node ptr hence no way to obtain the level...

Reply:
Seems like the unit test is potentially incorrect?

Reply:
They're all effed.  They partially initialize structures and try to run them through the system.  You change one little thing in the code and the unit tests blow up and you have to spend more time trying to figure that out and fixing it than it took to make the change...

Reply:
Is this related to the `HSE_REVISIT` you added to hse_meminfo() or whatever in kvdb_interface.c?

Reply:
Not sure what you're referring to, I didn't add any REVISTs to this patch... ???

Reply:
You mean my rant about unit tests in general?

Reply:
https://github.com/hse-project/hse/blob/master/lib/binding/kvdb_interface.c#L70

Reply:
Right.  The "mapi breaks initialization..." comments I left were because I ran into problems with functions that used the simple mocking method (intercepts the call and returns some prescribed error code).  That type of mocking breaks code that relies on the function having initialized the parameters being passed in before returning whatever it returns...